CRM/Civixero/Contact.php

```diff
@@ -190,7 +196,13 @@ public function push(array $params, int $limit = 10): bool {
         $locationTypeToSync = (int) Civi::settings()->get('xero_sync_location_type');
         $contact['email'] = $this->getPreferredEmail($locationTypeToSync, $record['contact_id']);
         $contact['phone'] = $this->getPreferredPhone($locationTypeToSync, $record['contact_id']);
-        $contact += $this->getPreferredAddress($locationTypeToSync, $record['contact_id']);
+        // Address is different to the other location fields because it has multiple fields.
+        // We might return NULL from getPreferredAddress which means "do not sync to Xero".
+        // That way we preserve any partial address that we might have in Xero and it will be synced next time it's pulled to Civi.
+        $contactAddress = $this->getPreferredAddress($locationTypeToSync, $record['contact_id']);
+        if ($contactAddress) {
+          $contact = array_merge($contact, $contactAddress);
+        }
```

Comment:
Per comment - we had some issues with existing (incomplete) addresses in Xero being overwritten. Passing empty keys overwrites Xero, passing nothing does not overwrite Xero.

Reply:
this seems fine